test/integration/test-visibility-states.js

```diff
@@ -80,11 +84,26 @@ t.run('Viewer Visibility State', () => {
         }
       }
 
+      function intersect() {
+        intersect_.apply(this, arguments);
```

Comment:
why this `this`?

Reply:
The `this` is the `Resources` instance, see `doPass()` too. sinon's `callsFake` does this so you can replace instance functions of objects more easily.

Reply:
Oh thats really cool. I didn't realize how that worked before. Thanks!